src/deploy/functions/release/fabricator.ts

```diff
@@ -337,12 +338,19 @@ export class Fabricator {
     }
 
     const mem = endpoint.availableMemoryMb || backend.DEFAULT_MEMORY;
-    if (mem >= backend.MIN_MEMORY_FOR_CONCURRENCY && endpoint.concurrency !== 1) {
-      await this.setConcurrency(
-        endpoint,
-        serviceName,
-        endpoint.concurrency || DEFAULT_GCFV2_CONCURRENCY
-      );
+    const hasCustomCPU = endpoint.cpu !== backend.memoryToGen1Cpu(mem);
```

Comment:
having a brain fart, but should this be comparing the `endpoint.cpu` to `memoryToGen2Cpu` instead?

In english words, I would've guessed that:

```
hasCustomCPU = endpoint.cpu does not match default settings for Gen 2 memory -> CPU mapping.
```

Reply:
I can change the name to be more clear or add a comment. This code is correct; gcf gen 1 is the "built in" memory that we get automatically and v2 is a default we use for our custom VM shapes.